lib/internal/streams/operators.js

```diff
@@ -118,27 +144,26 @@ function map(fn, options) {
             continue;
           }
 
-          if (typeof val?.catch === 'function') {
-            val.catch(onDone);
-          }
+          val = PromiseResolve(val);
+          cnt += 1;
+
 
-          queue.push(val);
+          queue.push(PromisePrototypeThen(val, onThen, onCatch));
```

Comment:
```suggestion
          PromisePrototypeCatch(val, onCatch);
          queue.push(PromisePrototypeThen(val, onThen));
```

Reply:
This will ensure that there is no unhandled rejection

Reply:
1. there is eslint error to not use `PromisePrototypeCatch`
2. and it doesn't work

Reply:
that should work, so we should try to understand why it doesn't work before creating a workaround